crml/staking/src/rewards/mod.rs

```diff
@@ -61,10 +58,12 @@ pub trait Trait: frame_system::Trait {
 	type TreasuryModuleId: Get<ModuleId>;
 	/// The number of historical eras for which tx fee payout info should be retained.
 	type HistoricalPayoutEras: Get<u16>;
-	/// The reward payouts would be split among several blocks when their number exceeds this threshold.
-	type PayoutSplitThreshold: Get<u32>;
+	/// The gap between subsequent payouts (in number of blocks)
+	type BlockPayoutInterval: Get<Self::BlockNumber>;
```

Comment:
It seems block spacing couples the proper behaviour of the rewards to HISTORY_DEPTH in the staking module in a sneaky way. At the time when we calculate and process payouts, ErasStakersClipped and ErasValidatorPrefs should still have the required data. However the history may have removed.

Reply:
history depth is for lazy payouts so that reward payouts can happen some time in future eras.
in this implementation we could remove it and have `LastEraStakers` only which seems much nicer solution.
From a public API perspective there's nothing wrong here, but we happen to know the internal details of the staking module and that it does it's own garbage cleaning so the storage could disappear if payout era is < current era + history depth.
we have integration tests for this reason